packages/inventory-compliance/src/SystemRulesTable.test.js

```diff
@@ -0,0 +1,157 @@
+
+const profileRules = [
+    {
+        system: 'aa9c4497-5707-4233-9e9b-1fded5423ef3',
+        profile: {
+            refId: 'xccdf_org.ssgproject.content_profile_pci-dss',
+            name: 'PCI-DSS v3 Control Baseline for Red Hat Enterprise Linux 7'
+        },
+        rules: [
+            {
+                title: 'Use direct-lvm with the Device Mapper Storage Driver',
+                severity: 'low',
+                rationale: 'foorationale',
+                refId: 'xccdf_org.ssgproject.content_rule_docker_storage_configured',
+                description: 'foodescription',
+                compliant: true,
+                identifier: {
+                    label: 'CCE-80441-9',
+                    system: 'https://nvd.nist.gov/cce/index.cfm',
+                    __typename: 'RuleIdentifier'
+                },
+                references: [],
+                __typename: 'Rule'
+            },
+            {
+                title: 'Enable the Docker service',
+                severity: 'medium',
+                rationale: 'foorationale',
+                refId: 'xccdf_org.ssgproject.content_rule_service_docker_enabled',
+                description: 'foodescription',
+                compliant: true,
+                identifier: {
+                    label: 'CCE-80440-1',
+                    system: 'https://nvd.nist.gov/cce/index.cfm',
+                    __typename: 'RuleIdentifier'
+                },
+                references: [],
+                __typename: 'Rule'
+            }
+        ]
+    }
+];
+
+describe('SystemRulesTable component', () => {
+    it('should render', () => {
+        global.insights = windowInsightsSpy;
+        remediationsApi.getResolutionsBatch.mockImplementation(
+            () => new Promise((callback) => callback(remediationsResponse))
+        );
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ false }
+                system={ system }
+            />
+        );
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it('should render a loading table', () => {
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ true }
+                system={ system }
+            />
+        );
+        expect(toJson(wrapper)).toMatchSnapshot();
+    });
+
+    it('should render filtered rows', () => {
+        global.insights = windowInsightsSpy;
+        remediationsApi.getResolutionsBatch.mockImplementation(
+            () => new Promise((callback) => callback(remediationsResponse))
+        );
+        const wrapper = shallow(
+            <SystemRulesTable
+                profileRules={ profileRules }
+                loading={ false }
+                system={ system }
+            />
+        );
+        const instance = wrapper.instance();
+        return instance.setInitialCurrentRows().then(() => {
```

Comment:
Please rewrite this to more streamlined approach. If one of these requests fails the entire test fails and we won't be able to find out which actually failed.

Reply:
:+1: - that can be done